internal/syncd/integrations/ecr/reconciler_test.go

```diff
@@ -0,0 +1,142 @@
+package ecr
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"ship-it/internal"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const inputYaml = `apiVersion: helmreleases.k8s.wattpad.com/v1alpha1
+kind: HelmRelease
+metadata:
+  creationTimestamp: null
+  name: example-microservice
+spec:
+  chart:
+    path: microservice
+    repository: wattpad.s3.amazonaws.com/helm-charts
+    revision: HEAD
+  releaseName: example-release
+  values:
+    image:
+      repository: 723255503624.dkr.ecr.us-east-1.amazonaws.com/bar
+      tag: baz
+status: {}
+`
+
+const expectedYaml = `apiVersion: helmreleases.k8s.wattpad.com/v1alpha1
+kind: HelmRelease
+metadata:
+  creationTimestamp: null
+  name: example-microservice
+spec:
+  chart:
+    path: microservice
+    repository: wattpad.s3.amazonaws.com/helm-charts
+    revision: HEAD
+  releaseName: example-release
+  values:
+    image:
+      repository: 723255503624.dkr.ecr.us-east-1.amazonaws.com/bar
+      tag: 78bc9ccf64eb838c6a0e0492ded722274925e2bd
+status: {}
+`
+
+type MockRepoService struct {
+	mock.Mock
+}
+
+func (m *MockRepoService) UpdateAndReplace(ctx context.Context, path string, image *internal.Image, msg string) error {
+	args := m.Called(ctx, path, image, msg)
+	return args.Error(0)
+}
+
+type MockIndexerService struct {
+	mock.Mock
+}
+
+func (m *MockIndexerService) Lookup(repo string) ([]types.NamespacedName, error) {
+	args := m.Called(repo)
+	return args.Get(0).([]types.NamespacedName), args.Error(1)
+}
+
+func TestReconcileLookupFailure(t *testing.T) {
+	mockRepoService := new(MockRepoService)
+	mockIndexService := new(MockIndexerService)
+	fakeReconciler := NewReconciler("foo/bar/resources", mockRepoService, mockIndexService)
```

Comment:
This isn't really a fake, it's a real reconciler that uses mock dependencies.

I've used the term `fake` in other places like [here](https://github.com/Wattpad/ship-it/pull/68/files#diff-83d733c3663e96ac502b181ebf9d47d4R34) to distinguish between a true mock and a fake implementation. The `fakeCache` there is a real implementation of a cache, no mocks, but it has limited functionality and it's purely in-memory to support unit testing.

I don't think we should call the thing we're testing "fake". It's very suggestive and makes it less clear what's actually being tested vs. the dependencies.

Reply:
Agreed, I will rename it to `mockReconciler`